### ISSUE TITLE
*.png Filter to export non-resource files is Ignored

### ISSUE THREAD

**Issue by Lolzorous** [godotengine/godot#60285]
### Godot version

Godot v3.4.4 stable

### System information

Windows 10

### Issue description

I have two filters set up to export non-resource files "*.png, *.gif" The gif filter works however the png filter does not and the original png's are still dropped leaving only the .imports when exporting the project.

I realize that deleting .png's and leaving a .import is working as intended I did read that bug report but I feel like the filter should work regardless

The reason why I want the .png and not the .stex file is because I'm using a plugin for the Discord bot api and I'm trying to get my bot to upload an image file message which works in editor since the png file path is still there but once I export it the path breaks. This is an example of the code I found to have my bot message an image:
![image](https://user-images.githubusercontent.com/103785070/163640979-1b1ef096-480a-4d92-b249-81d89a8a992c.png)
This particular example works mind you since again the .gif file exports no problem. I admittedly don't fully understand what's going on here but I know that the bytes var is the important part that give discord the image to load.

I have figured out a way to get the bot to post png's where I skip the whole file.open() part and instead I add a TextureRect with the image I want as a texture to my scene and use $TextureRect.texture.get_data() and then use the save_png_to_buffer() method to get the data required for discord to send the image but I feel like my filter to export .png's shouldn't be flat out ignored although I understand that this is a very specific case and I apologize if that makes this bug report a bit of a waste of your time.

### Steps to reproduce

The project I included is a scene with a file dialog popup that open as soon as you launch the scene.
The project it self has two files I put in it's res:// folder a HelloThankyou.png and a HelloThankyou.gif
Assuming export filters are saved with the project once it's exported and opened the File Dialog popup will show that there is a .gif file but the .png will not be there only it's .import counterpart. However the icon.png will be there for no other reason than to taunt me I presume.

### Minimal reproduction project

[ExportBugExample.zip](https://github.com/godotengine/godot/files/8498804/ExportBugExample.zip)

**Comment by Calinou** [comment]
This is intended, as PNG is normally considered a resource by Godot.

To avoid this, select PNG files which you don't intend to be imported by Godot in the FileSystem, go to the Import dock and set their import mode to **Keep (No Import)**.

**Comment by Lolzorous** [comment]
Oh thank you, I didn't realize that was an option. I played with the importer a little but all I really did was reimport is as an image to no avail. I'm still very new to godot & programming so thank you for the solution that is so much simpler then what I came up with. I very much appreciate your time and assistance.

I do just want to note though that my original thought process behind this was that .png's were a non-resource since they are ultimately not kept on export, but I'm realizing the distinction now that the filter works for any resource that godot doesn't import.
